en/components/general/wpf_to_angular_guide/angular_events.md

```diff
@@ -0,0 +1,142 @@
+---
+title: Angular Events
+_description: Learn how to respond to events in Angular and implement your own events.
+_keywords: WPF, Windows Presentation Foundation, Angular, Ignite UI for Angular, Events, Event Binding
+---
+
+## Responding to Events
+
+Every app you have probably ever had to write in your lifetime had to respond to some type of user interaction one way or another. The most common way to do that is with some type of event system. In WPF we had routed events, CLR events and commands while In Angular we have DOM events.
+
+Here is a simple example how you respond to a click event of a button in WPF:
+
+```xml
+<Button Click="Button_Click">Click Me</Button>
+```
+```csharp
+private void Button_Click(object sender, RoutedEventArgs e)
+{
+    Console.WriteLine("Hello World");
+}
+```
+
+The same thing in Angular would look like this:
+```html
+<button (click)="onClicked()">Click Me</button>
+```
+```typescript
+onClicked() {
+    console.log('Hello World');
+}
+```
+
+In WPF we are used to getting information about the event, such as the sender or any event arguments. In Angular we can use the `$event` variable. This variable will provide information to us about the event that was invoked.
+
+```html
+<button (click)="onClicked($event)">Click Me</button>
+```
+```typescript
+onClicked(event) {
+    console.log('Hello World');
+    console.log(event.target);
+}
+```
+
+Sometimes passing the event object might not be very useful. Instead you may want to pass for example the value of an `input` on the page.
+
+```html
+<input #messageInput>
+<button (click)="onClicked(messageInput.value)">Click Me</button>
+```
+```typescript
+onClicked(message) {
+    console.log(message);
+}
+```
+
+Let us say that we want to print the value of an input on pressing Enter. You could do that in Angular like this:
+
+```html
+<input #messageInput (keyup)="onInputKeyup($event, messageInput.value)">
+```
+```typescript
+onInputKeyup(event, message) {
+    if (event.keyCode === 13) {
+        console.log(message);
+    }
+}
+```
+
+However, in Angular there is an easier way to do that, you could bind to the `keyup.enter` pseudo-event. This way the event handler will be called only when the user presses Enter.
+
+```html
+<input #messageInput (keyup.enter)="onInputKeyup(messageInput.value)">
+```
+```typescript
+onInputKeyup(message) {
+    console.log(message);
+}
+```
+
+## Responding to Events of a Component
+
+In WPF often when you define your own reusable controls, you need to listen for an event of that control inside it. Most commonly you will do this by overriding a base method like this:
```

Comment:
Changed to:
"In WPF, when you create your own custom controls, often you need to extend or modify some base events like this:"

I would like to keep "often" since it is not always required to override events.